runtime/onert/core/src/compiler/pass/ConstantInsertionPass.cc

```diff
@@ -40,9 +40,25 @@ void ConstantInsertionPass::callback(const ir::OperationIndex &node_index, ir::I
     auto &object = _graph.operands().at(input);
 
     const auto key = ReplaceKey{input, factor};
-    if (object.isConstant() && (object.getUses().size() >= 2 ||
-                                _replace_operands_map.find(key) != _replace_operands_map.end()))
+
+    // Skip if the operand is not constant or has only one use
+    if (!object.isConstant() || object.getUses().size() < 2)
+      continue;
+
+    // 1st use of multi-used constant operand.
+    // Register original operand into keep_operand map for later reuse
+    if (_keep_operands_map.find(input) == _keep_operands_map.end())
+    {
+      _keep_operands_map.emplace(input, factor);
+      continue;
+    }
+
+    // Same backend and layout with original operand. Keep using original operand
+    if (_keep_operands_map.at(input) == factor)
+      continue;
+
```

Comment:
(optional)

Here is my trial to use `insert`'s return value — [`std::pair<iterator, bool>`](http://en.cppreference.com/w/cpp/utility/pair).

It is just for recording. I am not sure the code is better.

### 1st

`.second` seems not readable.

```suggestion
    if (_keep_operands_map.insert({input, factor}).second || _keep_operands_map.at(input) == factor)
      continue;
```

### 2nd

Thus, I've used c++17 features:
- if statements with initializer
- structured binding

```suggestion
    if (auto const [existing, ok] = _keep_operands_map.insert({input, factor}); ok || existing->second == factor)
      continue;
```

It requires an update in `CMakeLists.txt`.

```
set(CMAKE_CXX_STANDARD 17)
```